compiler-cli/src/lsp.rs

```diff
@@ -225,8 +226,19 @@ impl LanguageServer {
         let warnings = self.compiler.project_compiler.take_warnings();
         for warn in warnings {
             let diagnostic = warn.to_diagnostic();
-            match to_lsp_diagnostic(diagnostic) {
-                Some((path, diagnostic)) => self.store_diagnostic(path, diagnostic),
+            match to_lsp_diagnostic(diagnostic.clone()) {
+                Some((path, lsp_diagnostic)) => {
+                    self.store_diagnostic(path.clone(), lsp_diagnostic.clone());
+
+                    if let Some(hint) = diagnostic.hint {
+                        let lsp_hint = Diagnostic {
+                            severity: Some(DiagnosticSeverity::HINT),
+                            message: hint,
+                            ..lsp_diagnostic
+                        };
```

Comment:
I think this logic should live within the `store_diagnostic` function

Reply:
I'm not sure, because `stone_diagnostic` just receives the diagnostics and store it. I think this function shouldn't have any split logic.

Reply:
Currently hints are only surfaced for warnings, we need them to always be included in the LSP output but we also don't want to have to copy this bit of code to every place that the `store_diagnostic` function is called. It's not ever correct to do one without the other so they should be the same function.